server/accounts.go

```diff
@@ -3125,6 +3125,13 @@ func buildInternalNkeyUser(uc *jwt.UserClaims, acts map[string]struct{}, acc *Ac
 
 const fetchTimeout = 2 * time.Second
 
+func FetchAccount(res AccountResolver, name string) (string, error) {
```

Comment:
From my look at references in server code for Fetch() I see that we call it 4 times and here you have replaced 3 calls. Is that intentional that one is not replaced?

If the answer is that you missed one and they should all be replaced, then why not update Fetch() directly? Is it because this function is exposed and you don't want to break usage? It should be used only by the server, so should not be a problem. If you were expecting to have replaced all calls if Fetch to FetchAccount but don't want to break the old behavior of Fetch(), it could be an option of the account resolver. So all Fetch() would either verify or not based on that option.

Reply:
I assume the extra fetch is part of a test? (see my search results below)
I have not changed tests to keep this change small. (the presumption is that fetch in a test will be using the correct names).

I wrap the call to `res.Fetch` inside of `FetchAccount` so the code to check the key does not need to repeat in each of the account resolver individually.

Method specification
    Fetch
Found usages  (4 usages found)
    nats-server  (4 usages found)
        server  (2 usages found)
            accounts.go  (1 usage found)
                FetchAccount  (1 usage found)
                    3132 return res.Fetch(name)
            events_test.go  (1 usage found)
                Fetch  (1 usage found)
                    2075 return sr.AccountResolver.Fetch(name)
        test  (2 usages found)
            leafnode_test.go  (2 usages found)
                TestLeafNodeSignerUser  (2 usages found)
                    1581 token, err := s.AccountResolver().Fetch(apk1)
                    1604 tt, err := s.AccountResolver().Fetch(apk1)

Reply:
There is another call to Fetch() but I realize now that we pass "", which is just to check if the account resolver is reachable.

Reply:
Since so far this only used in the server package, does this need to be exposed? Could it be fetchAccount() instead?

Reply:
Absolutely, will make that change. Had it public essentially because Fetch was public.